examples/private/main.tf

```diff
@@ -0,0 +1,224 @@
+resource "aws_security_group" "client_sec_group" {
+  name        = "${var.deployment_prefix}-client-sg"
+  description = "client security group"
+  vpc_id      = var.vpc_id
+  tags        = var.tags
+
+  ingress {
+    description = "Allow oubound to ssh"
+    from_port   = 22
+    to_port     = 22
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+}
+module "redpanda-cluster" {
+  source                          = "../../"
+  public_key_path                 = var.public_key_path
+  nodes                           = var.nodes
+  deployment_prefix               = var.deployment_prefix
+  enable_monitoring               = var.enable_monitoring
+  tiered_storage_enabled          = var.tiered_storage_enabled
+  allow_force_destroy             = var.allow_force_destroy
+  distro                          = var.distro
+  hosts_file                      = var.hosts_file
+  tags                            = var.tags
+  clients                         = 1
+  security_groups_client          = [aws_security_group.client_sec_group.id]
+  associate_public_ip_addr        = true
+  associate_public_ip_addr_client = true
+  availability_zone               = ["us-west-2a"]
+  egress_rules                    = {
+    "SSH" = {
+      description     = "Allow oubound to ssh"
+      from_port       = 22
+      to_port         = 22
+      protocol        = "tcp"
+      enabled         = true
+      cidr_blocks     = ["0.0.0.0/0"]
+      self            = null
+      security_groups = [aws_security_group.client_sec_group.id]
+    }
+  }
+  ingress_rules = {
+    "SSH" = {
+      description     = "Allow inbound to ssh"
+      from_port       = 22
+      to_port         = 22
+      protocol        = "tcp"
+      enabled         = true
+      cidr_blocks     = ["0.0.0.0/0"]
+      self            = null
+      security_groups = [aws_security_group.client_sec_group.id]
+    }
+    "Kafka" = {
+      description     = "Allow anywhere inbound to access the Redpanda Kafka endpoint"
+      from_port       = 9092
+      to_port         = 9092
+      protocol        = "tcp"
+      enabled         = true
+      cidr_blocks     = null
+      self            = true
+      security_groups = [aws_security_group.client_sec_group.id]
+    }
+    "RPC" = {
+      description     = "Allow security-group only to access Redpanda RPC endpoint for intra-cluster communication"
+      from_port       = 33145
+      to_port         = 33145
+      protocol        = "tcp"
+      enabled         = true
+      cidr_blocks     = null
+      self            = true
+      security_groups = [aws_security_group.client_sec_group.id]
+    }
```

Comment:
Question out of terraform ignorance:

You mention: `Allow anywhere inbound to access the Redpanda Kafka endpoint` and set the security groups, and say: `Allow security-group only to access Redpanda RPC endpoint for intra-cluster communication` with the same configuration.

Aren't both the same? aren't both `Allow security-group only to access Redpanda <Kafka Endpoint/RPC endpoint>`?

Reply:
Ah shoot I forgot to update the descriptions. They're all copy pasted from the module so that I could associate them with the client SG, but some of them have been changed from all open to just within the server SG for hardening purposes.